packages/flutter/test/material/app_bar_test.dart

```diff
@@ -3167,6 +3167,36 @@ void main() {
     expect(titleOffset.dx, backButtonOffset.dx + titleSpacing);
   });
 
+  testWidgets('AppBar actions padding can be adjusted', (WidgetTester tester) async {
+    const double testActionsPadding = 8.0;
+    final Key actionKey = UniqueKey();
+    final Finder finder = find.byKey(actionKey);
+
+    Widget buildApp([EdgeInsetsGeometry? actionsPadding]) {
+      return MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(
+            actions: <Widget>[
+              SizedBox.square(key: actionKey, dimension: 40.0),
+            ],
+            actionsPadding: actionsPadding,
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildApp());
+
+    final Offset offsetA = tester.getTopRight(finder);
+
+    // use theoretical Material 3 default
+    await tester.pumpWidget(buildApp(const EdgeInsets.only(right: testActionsPadding)));
+
+    final Offset offsetB = tester.getTopRight(finder);
+
+    expect(offsetB.dx, equals(offsetA.dx - testActionsPadding));
```

Comment:
`actionsPadding` value can be anything, It doesn't have to be 'theoretical Material 3 default" 

First test without `actionsPadding` and the provide a custom padding